src/url-params.ts

```diff
@@ -189,16 +195,29 @@ export class AnalyticsUrlParams {
 			actualDomainMatch = null;
 		}
 
-		const destinationDomainMatch = destinationUrl
-			? destinationUrl.hostname.match(regex)
-			: null;
+		actualDomain = actualDomainMatch ? actualDomainMatch.toString() : null;
+		if (typeof destinationUrl === 'string') {
+			try {
+				destinationUrl = new URL(destinationUrl);
+				destinationDomainMatch = destinationUrl
+					? destinationUrl.hostname.match(regex)
+					: null;
+
+				destinationDomain = destinationDomainMatch
+					? destinationDomainMatch.toString()
+					: null;
+			} catch (err) {
+				return ''; //I dont know what we should do when an error like this happens. Any ideas??
+			}
+		} else {
+			destinationDomainMatch = destinationUrl
```

Comment:
```suggestion
			const destinationDomainMatch = destinationUrl
```
> Just declare it in this scope, since this is only used in this limited scope as a temp variable and its value isn't useful outside this context.